modin/core/dataframe/algebra/groupby.py

```diff
@@ -124,10 +127,10 @@ def map(
         # present in the func-dict:
         apply_func = cls.try_filter_dict(map_func, df)
         if other is not None:
-            # Other is a broadcasted partition that represents 'by' columns
-            # Concatenate it with 'df' to group on its columns names
-            other = other.squeeze(axis=axis ^ 1)
-            if isinstance(other, pandas.DataFrame):
+            # Other is a broadcasted partition that represents 'by' data to group on.
+            # If 'drop' then the 'by' data came from the 'self' frame, thus
+            # inserting missed columns to the partition to group on them.
+            if drop or isinstance(other := other.squeeze(axis=1), pandas.DataFrame):
```

Comment:
this previously was handled incorrectly which caused 'by' columns to be aggregated ([they were dropped afterward](https://github.com/modin-project/modin/blob/2ebc9cf51bfc773e3d4c898f5a33c0f60ad7ebc5/modin/core/dataframe/algebra/groupby.py#L195-L200) so no effect to result), the 'skew' aggregation doesn't tolerate non-numeric columns ('by') to be aggregated thus this correction is required for proper behavior.

Reply:
Why was `squeeze(axis=axis ^ 1)` changed to `squeeze(axis=1)`?

Reply:
my bad, reverted to `axis ^ 1`

this was not alerted by tests because we only support groupby along '0' axis for now